### PR TITLE
Routing

### PR DESCRIPTION
**Pull request by namratachaudhary**
- [ ] Setup
- [ ] middlewares
- [ ] basic handlers

### CODE REVIEW

File: src/leaflike/routes.clj [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+            [ring.util.response :as res]))
+
+(defn yay [request]
+  (res/response "Yay!"))
+
+(defn welcome [request]
+  (res/response "Welcome to leaflike"))
+
+(def urls
+  {{:get "" welcome
```
Comment (yudistrange) [inline src/leaflike/routes.clj]: This seems like a bug.

File: src/leaflike/db.clj [hunk 2]
```diff
@@ -0,0 +1,4 @@
+(ns leaflike.db
+  (:require [clojure.java.jdbc :as jdbc]))
+
+(defn db-spec [] {:connection-uri "jdbc:postgresql://localhost:5432/leaflike"})
```
Comment (yudistrange) [inline src/leaflike/db.clj]: This is more of a discussion: If your function is returning the same result for every call does it make sense for it to be a value instead?

File: src/leaflike/server.clj [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
-  (:require [org.httpkit.server :as httpkit]))
+  (:require [org.httpkit.server :as httpkit]
+            [ring.middleware.json :refer [wrap-json-body]]
+            [leaflike.routes :as routes]))
 
-(defn app [req]
-  {:status  200
-   :headers {"Content-Type" "application/json"}
-   :body    "Welcome to Leaflike"})
+(defn app []
```
Comment (yudistrange) [inline src/leaflike/server.clj]: The same defn vs def discussion is applicable here as well.

File: src/leaflike/routes.clj [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+            [ring.util.response :as res]))
+
+(defn welcome [_]
+  (res/response "Welcome to leaflike"))
+
+(defn yay [_]
+  (res/response "Yay!"))
+
+(defn ping [request]
+  (res/response {:ping (-> request :route-params :ping)}))
```
Comment (yudistrange) [inline src/leaflike/routes.clj]: Just FYI this will **not** return a valid JSON.

Comment (namratachaudhary) [reply src/leaflike/routes.clj]: yes, we need to add the middleware to return the valid json.

File: src/leaflike/server.clj (hunk 3 above)
Comment (namratachaudhary) [reply src/leaflike/server.clj]: was following convention. Your suggestion sounds about right too.

**Pull request merged**